reference/enterprise-api-ref/cmdb-api.markdown

```diff
@@ -0,0 +1,317 @@
+                }
+            }
+        }
+    },
+    "meta": {
+        "count": "1",
+        "page": 1,
+        "timestamp": 1619116399,
+        "total": "1",
+        "most_recent_cmdb_change": "13"
+    }
+}
+```
+
+## Get configuration by hostkey
+
+**URI:** https://hub.cfengine.com/cmdb/:hostkey
+
+**Method:** GET
+
+**Parameters:**
+* **hostkey** *(string)*
+  Unique host identifier.
+
+**Example request (curl):**
+
+```
+curl -k --user <username>:<password> \
+  -X GET \
+  https://hub.cfengine.com/api/cmdb/SHA=f622992fa4525070f47da086041a38733496f03a77880f70b1ce6784c38f79ab 
+```
+
+**Example response:**
+
+```
+HTTP 200 Ok
+{
+    "hostkey": "SHA=f622992fa4525070f47da086041a38733496f03a77880f70b1ce6784c38f79ab",
+    "value": {
+        "classes": {
+            "My_class": {},
+            "My_class2": {
+                "comment": ""
+            }
+        },
+        "variables": {
+            "HubCMDB:My.hostname": {
+                "value": "host1.cfengine.com",
+                "comment": "My hostname should be set to this"
+            },
+            "Namespace:BundleName.VariableName": {
+                "value": "myvalue"
+            }
+        }
+    }
+}
+```
+
+## Create new configuration
+
+**URI:** https://hub.cfengine.com/cmdb
+
+**Method:** POST
+
+**Parameters:**
+* **hostkey** *(string)*
+  Unique host identifier.
+* **classes** *(JSON object)*
+  The format is a JSON object where the key is class name and value is another JSON object
+  with an optional `comment` property.
+  Example:
+```
+{
+   "classes":{
+      "My_class":{
+         
+      },
+      "My_class2":{
+         "comment":"comment body"
+      }
+   }
+}
+```
+  
+* **variables** *(JSON object)*
+  The format is a JSON object where the key is variable name and value is another JSON object
+  with a required `value` property and an optional `comment` property.
+  Example:
+```
+{
+   "variables":{
+      "Namespace:BundleName.VariableName":{
+         "value":"myvalue"
+      },
+      "HubCMDB:My.hostname":{
+         "value":"host1.cfengine.com",
+         "comment":"My hostname should be set to this"
+      }
+   }
+}
+```
+
+
+**Example request (curl):**
+
+```
+curl -k --user <username>:<password> \
+  -X POST \
+  https://hub.cfengine.com/api/cmdb \
+  -H 'content-type: application/json' \
+  -d '{
+   "hostkey":"SHA=f622992fa4525070f47da086041a38733496f03a77880f70b1ce6784c38f79ab",
+   "classes":{
+      "My_class2":{
+         "comment" : ""
+      },
+      "My_class": {}
+   },
+   "variables":{
+      "Namespace:BundleName.VariableName":{
+         "value":"myvalue"
+      },
+      "HubCMDB:My.hostname":{
+         "value":"host1.cfengine.com",
+         "comment":"My hostname should be set to this"
+      }
+   }
+}'
+```
+
+**Example response:**
+
+```
+HTTP 201 Created
+```
+
+## Update configuration
+
+**URI:** https://hub.cfengine.com/cmdb/:hostkey
+
+**Method:** PATCH
+
+**Parameters:**
+* **hostkey** *(string)*
```

Comment:
same as above, it seems a bit confusing to me to include the classes and variables "keys" in the examples when the value for those parameters is really just the internal json structure instead. So maybe remove the key part in the snippets like:

classes example:
```
{
  "My_class": {},
  "My_class2":{
    "comment":"comment body"
  }
}
```

Reply:
In order to remove the variables and classes keys from the POST I think we would have to then have 2 different endpoints, one for vars and one for classes.

Reply:
I wasn't saying remove it from the data sent to the end-point, but rather remove it from the snippets and then put it together later. It was confusing to me.

Reply:
re-reading it now it makes sense, but initially it was confusing. Maybe ask another non-experienced person to review this?! @aleksandrychev

Reply:
ping @olehermanse  @chiachenglu